probe.go

```diff
@@ -0,0 +1,325 @@
+		return fmt.Sprintf("zones/%s/rrsets/%s/%s/probes", zone, typ, name)
+	}
+	return fmt.Sprintf("zones/%s/rrsets/%s/%s/probes/%s", zone, typ, name, guid)
+}
+
+// AlertPath generates the URI path for an alert
+func AlertPath(zone, typ, name string) string {
+	return fmt.Sprintf("zones/%s/rrsets/%s/%s/alerts", zone, typ, name)
+}
+
+func probeQueryPath(zone, typ, name, query string) string {
+	uri := ProbePath(zone, typ, name, "")
+	if query != "" {
+		uri = fmt.Sprintf("%s?sort=NAME&query=%s", uri, query)
+	}
+	return uri
+}
+
+func probeAlertQueryPath(zone, typ, name string, offset int) string {
+	baseURI := AlertPath(zone, typ, name)
+	return fmt.Sprintf("%s?offset=%d", baseURI, offset)
+}
+
+// ListAllProbeAlerts returns all probe alerts with name, type & zone
+func (s *SBTCService) ListAllProbeAlerts(name, typ, zone string) ([]ProbeAlertDataDTO, error) {
+	// TODO: Sane Configuration for timeouts / retries
+	maxerrs := 5
+	waittime := 5 * time.Second
+
+	// init accumulators
+	as := []ProbeAlertDataDTO{}
+	offset := 0
+	errcnt := 0
+
+	for {
+		reqAlerts, ri, res, err := s.ListProbeAlerts(name, typ, zone, offset)
+		if err != nil {
+			if res.StatusCode >= 500 {
+				errcnt = errcnt + 1
+				if errcnt < maxerrs {
+					time.Sleep(waittime)
+					continue
+				}
+			}
+			return as, err
+		}
+
+		log.Printf("ResultInfo: %+v\n", ri)
+		for _, a := range reqAlerts {
+			as = append(as, a)
+		}
+		if ri.ReturnedCount+ri.Offset >= ri.TotalCount {
+			return as, nil
+		}
+		offset = ri.ReturnedCount + ri.Offset
+		continue
+	}
+}
+
+// ListProbeAlerts returns the probe alerts with name, type & zone, accepting an offset
+func (s *SBTCService) ListProbeAlerts(name, typ, zone string, offset int) ([]ProbeAlertDataDTO, ResultInfo, *Response, error) {
+	var ald ProbeAlertDataListDTO
+
+	uri := probeAlertQueryPath(zone, typ, name, offset)
+	res, err := s.client.get(uri, &ald)
+
+	as := []ProbeAlertDataDTO{}
+	for _, a := range ald.Alerts {
+		as = append(as, a)
+	}
+	return as, ald.Resultinfo, res, err
+}
+
+// GetProbe returns a probe with name, type, zone & guid
+func (s *SBTCService) GetProbe(name, typ, zone, guid string) (ProbeInfoDTO, *Response, error) {
+	var t ProbeInfoDTO
+	uri := ProbePath(zone, typ, name, guid)
+	res, err := s.client.get(uri, &t)
+	return t, res, err
+}
+
+// CreateProbe creates a probe with name, type & zone using the ProbeInfoDTO dp
+func (s *SBTCService) CreateProbe(name, typ, zone string, dp ProbeInfoDTO) (*Response, error) {
+	uri := ProbePath(zone, typ, name, "")
+	var ignored interface{}
+	return s.client.post(uri, dp, &ignored)
+}
+
+// UpdateProbe updates a probe given a name, type, zone & guid with the ProbeInfoDTO dp
+func (s *SBTCService) UpdateProbe(name, typ, zone, guid string, dp ProbeInfoDTO) (*Response, error) {
+	uri := ProbePath(zone, typ, name, guid)
+	var ignored interface{}
+	return s.client.put(uri, dp, &ignored)
+}
+
+// ListProbes returns all probes by name, type & zone, with an optional query
```

Comment:
Perhaps we should mention that there isn't a `ListAllProbes` because this API doesn't support pagination.

Or maybe we should still have one to wrap the error handling in other `ListAll*` functions?